src/components/table/Table.js

```diff
@@ -1,28 +1,417 @@
+  },
+  table: {
+    minWidth: 500
+  },
+  tableWrapper: {
+    overflowX: 'auto'
+  }
+}));
+
+// Pagination Style
+const usePaginationStyles = makeStyles(theme => ({
+  root: {
+    flexShrink: 0,
+    color: theme.palette.text.secondary,
+    marginLeft: theme.spacing(10)
+  }
+}));
+
+// Filter Style
+const useFilterStyles = makeStyles(theme => ({
+  root: {
+    display: 'flex',
+    flexWrap: 'wrap'
+  },
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 250,
+    maxWidth: 300
+  }
+}));
+
+const TablePaginationActions = props => {
+  const paginationClasses = usePaginationStyles();
+  const theme = useTheme();
+  const { count, page, rowsPerPage, onChangePage } = props;
+
+  const handleFirstPageButtonClick = event => {
+    onChangePage(event, 0);
+  };
+
+  const handleBackButtonClick = event => {
+    onChangePage(event, page - 1);
+  };
+
+  const handleNextButtonClick = event => {
+    onChangePage(event, page + 1);
+  };
+
+  const handleLastPageButtonClick = event => {
+    onChangePage(event, Math.max(0, Math.ceil(count / rowsPerPage) - 1));
+  };
+
+  return (
+    <div className={paginationClasses.root}>
+      <IconButton
+        onClick={handleFirstPageButtonClick}
+        disabled={page === 0}
+        aria-label='First Page'
+      >
+        {theme.direction === 'rtl' ? <LastPageIcon /> : <FirstPageIcon />}
+      </IconButton>
+      <IconButton
+        onClick={handleBackButtonClick}
+        disabled={page === 0}
+        aria-label='Previous Page'
+      >
+        {theme.direction === 'rtl' ? (
+          <KeyboardArrowRight />
+        ) : (
+          <KeyboardArrowLeft />
+        )}
+      </IconButton>
+      <IconButton
+        onClick={handleNextButtonClick}
+        disabled={page >= Math.ceil(count / rowsPerPage) - 1}
+        aria-label='Next Page'
+      >
+        {theme.direction === 'rtl' ? (
+          <KeyboardArrowLeft />
+        ) : (
+          <KeyboardArrowRight />
+        )}
+      </IconButton>
+      <IconButton
+        onClick={handleLastPageButtonClick}
+        disabled={page >= Math.ceil(count / rowsPerPage) - 1}
+        aria-label='Last Page'
+      >
+        {theme.direction === 'rtl' ? <FirstPageIcon /> : <LastPageIcon />}
+      </IconButton>
+    </div>
+  );
+};
+
+TablePaginationActions.propTypes = {
+  count: PropTypes.number.isRequired,
+  onChangePage: PropTypes.func.isRequired,
+  page: PropTypes.number.isRequired,
+  rowsPerPage: PropTypes.number.isRequired
+};
+
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250
+    }
+  }
+};
+
+const SimpleTable = ({
+  config,
+  colConfig,
+  data,
+  page,
+  rowsPerPage,
+  totalRows,
+  onPaginate,
+  filterSelects,
+  onFilter
+}) => {
+  const tableClasses = useTableStyles();
+  const FilterClasses = useFilterStyles();
+
+  const [filters, setFilters] = useState({
```

Comment:
Why not

```js
const [filters, setFilters] = useState(_.transform(colConfig, (initialFilter, value, key) => {
  initialFilter[key] = [];
  return initialFilter;
}, {}));
```

Reply:
Fix this to:
const [filters, setFilters] = useState(
    _.transform(
      colConfig,
      (initialFilter, config) => {
        initialFilter[config.id] = [];
        return initialFilter;
      },
      {}
    )
  );